tests/assembly/stack-protector/stack-protector-heuristics-effect.rs

```diff
@@ -10,6 +10,10 @@
 //@ [none] compile-flags: -Z stack-protector=none
 //@ compile-flags: -C opt-level=2 -Z merge-functions=disabled
 //@ min-llvm-version: 17.0.2
+//@ ignore-test
+// FIXME: The LLVM stack protector code assumes that alloca types are meaningful,
+// so using [n x i8] types causes it to emit stack protection code for all allocas.
+// It needs to be changed to use some meaningful heuristic.
```

Comment:
This heuristic should be in the frontend rather than looking at alloca types, it'll require LLVM changes

Reply:
Judging from https://github.com/rust-lang/rust/issues/114903#issuecomment-1761828804, and the fact that we're only going to stabilize `stack-protector=all` [right now](https://github.com/rust-lang/rust/pull/121742#issuecomment-1969197171), this issue seems to be well known, so making this change shouldn't be an issue.